### DISCUSSION TITLE
Add support for SQLite's foreign key

### DISCUSSION THREAD

**Discussion by ad-si** [azimuttapp/azimutt#20] [Features]
https://www.sqlite.org/foreignkeys.html

**Comment by loicknuchel** [comment]
Yep!
As there is a lot of specificities in SQL languages, I'm doing what "is needed" => what I need or people ask ;)
I will add this to my todo list :)

BTW: do you have some experience with Elm parser ?
Today the parser is built with regex, this was "easy" but I hit a few walls ^^

**Reply by ad-si** [reply]
> BTW: do you have some experience with Elm parser ?

I actually do 😅. I just wrote a parser for a custom schema specification language. It causes some head scratching in the beginning, but it definitely scales better than Regex. The trick is to force yourself to start with the atoms and then work your way up to the full AST.

**Reply by loicknuchel** [reply]
Oh, that's awesome ;)
I already tried parsers in several other languages but it has always been hard...
Would you be interested in some pair programming to get me started with it ?

**Reply by ad-si** [reply]
Ah sorry, a lot of stuff came up at work and it will be some time before I can think of anything else 😭

**Reply by loicknuchel** [reply]
Outch, be strong!
No problem, I'm moving forward on many part and will do what I can (or have time for ^^)
Tell me if you are interested and have some time to do some pair one day ;)

**Comment by loicknuchel** [comment]
Hi @ad-si 

I implemented SQLite foreign keys (in https://github.com/azimuttapp/azimutt/pull/41).
Can you try it and tell me if this is fine for you ?

Thanks,
Cheers!

**Reply by ad-si** [reply]
Uhm, didn't get that far 😅 => https://github.com/azimuttapp/azimutt/issues/43

**Reply by loicknuchel** [reply]
haha, will check ^^
